### PR TITLE
Megafauna once again gibs immortal miners

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Memento Mori + Megafauna no longer equals softlocking the game.

## Why It's Good For The Game

Miners once again will be gibbed. 

## Changelog
:cl:
fix: Megafauna once again gibs immortal miners
/:cl:

### CODE REVIEW

**Comment by XDTM** [comment]
Have you tested this? Because reading the code this should change nothing at all, since the `if` is below the `stat != DEAD` check

**Comment by SmArtKar** [comment]
> Have you tested this? Because reading the code this should change nothing at all, since the `if` is below the `stat != DEAD` check

aw fuck thanks man. It should go without else and one tab higher

**Comment by ATH1909** [comment]
Isn't being able to be stuck in endless torment that you WISH you could die to (but can't) one of the defining features of memento mori?

**Comment by XDTM** [comment]
It's not really the core theme of the item, and it kinda just sucks in a gameplay sense to get stuck not being able to do anything besides ghosting.
Plus, it makes sense to allow this since memento mori users are not immune to gibbing in the first place, it's just megafauna AI not being prepared for this situation.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -120,6 +120,9 @@
 		if(L.stat != DEAD)
 			if(!client && ranged && ranged_cooldown <= world.time)
 				OpenFire()
+
+			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:124]: Does gib() go through TRAIT_NODEATH?

Comment (XDTM) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:124]: Yes, anything that calls the death() proc directly does

**Pull request merged**